In [ ]:
url = 'https://www.amazon.in/Good-knight-Vapourizer-Mosquito-Repellent/dp/B01N5A5TVN/ref=cm_cr_arp_d_product_top?ie=UTF8&th=1'

In [ ]:
import pandas as pd
df = pd.read_csv('/content/good_night_reviews.csv')

In [ ]:
df.shape

(601, 5)

In [ ]:
df.isna().sum()

Reviewer    0
Rating      0
Title       0
Date        0
Body        0
dtype: int64

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 601 entries, 0 to 600
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Reviewer  601 non-null    object 
 1   Rating    601 non-null    float64
 2   Title     601 non-null    object 
 3   Date      601 non-null    object 
 4   Body      601 non-null    object 
dtypes: float64(1), object(4)
memory usage: 23.6+ KB


In [ ]:
df['Date'] = pd.to_datetime(df['Date'], format='mixed', dayfirst=True, errors='coerce')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 601 entries, 0 to 600
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype         
---  ------    --------------  -----         
 0   Reviewer  601 non-null    object        
 1   Rating    601 non-null    float64       
 2   Title     601 non-null    object        
 3   Date      601 non-null    datetime64[ns]
 4   Body      601 non-null    object        
dtypes: datetime64[ns](1), float64(1), object(3)
memory usage: 23.6+ KB


In [ ]:
df

Reviewer  Rating  \
0                   ROSHAN     5.0   
1          P.Kishore Kumar     3.0   
2                   Vikram     1.0   
3                   Devika     3.0   
4           Mohammad Azhar     1.0   
..                     ...     ...   
596           Lamp Abience     5.0   
597          Ranabir Ghosh     5.0   
598  SVV SATYANARANARAYANA     5.0   
599          Nusrat jabeen     5.0   
600        Ashu Vashishtha     5.0   

                                                 Title       Date  \
0                                                 Good 2025-06-27   
1                                              Nothing 2025-06-27   
2                              Do not buy this product 2025-06-21   
3                                                 Good 2025-06-16   
4    Good night refill stick is broken. Lost of one... 2025-06-10   
..                                                 ...        ...   
596                                          Skin care 2023-08-03   
597                                     Does it's work 2023-08-02   
598                                       Nice product 2023-07-24   
599                                               Good 2023-07-18   
600                       Amazing Product Price Saving 2023-07-17   

                                                  Body  
0                                                 Good  
1                                                 Good  
2    Not happy with product quality.\nVery bad smel...  
3                             For mosquito repellent….  
4    Product coil broken, how to replace in refill?...  
..                                                 ...  
596  Click to play video\nThis good night is very p...  
597  It's useful for a small bedroom say 10ft by 12ft.  
598                best for mosquitos. value for money  
599                                               Good  
600                                I bought it for 360  

[601 rows x 5 columns]

In [ ]:
df['Date'].apply(lambda x: isinstance(x, pd.Timestamp)).all()

np.True_

In [ ]:
df["Body"][2]

"Not happy with product quality.\nVery bad smell. I am using good knight from last 10 years, but it's looks like different product.\nAll refills are comming out of shipping box as packing inside the company box is not done."

In [ ]:
text = "Not happy with product quality.\nVery bad smell. I am using good knight from last 10 years, but it's looks like different product.\nAll refills are comming out of shipping box as packing inside the company box is not done."

In [ ]:
def nremoval(text):
  return ' '.join(text.split('\n'))

In [ ]:
nremoval(text)

"Not happy with product quality. Very bad smell. I am using good knight from last 10 years, but it's looks like different product. All refills are comming out of shipping box as packing inside the company box is not done."

In [ ]:
cols = ['Reviewer', 'Title', 'Body']
for i in cols:
  df[i] = df[i].apply(nremoval)

In [ ]:
df.head()

Reviewer  Rating                                              Title  \
0           ROSHAN     5.0                                               Good   
1  P.Kishore Kumar     3.0                                            Nothing   
2           Vikram     1.0                            Do not buy this product   
3           Devika     3.0                                               Good   
4   Mohammad Azhar     1.0  Good night refill stick is broken. Lost of one...   

        Date                                               Body  
0 2025-06-27                                               Good  
1 2025-06-27                                               Good  
2 2025-06-21  Not happy with product quality. Very bad smell...  
3 2025-06-16                           For mosquito repellent….  
4 2025-06-10  Product coil broken, how to replace in refill?...

In [ ]:
df = df.drop_duplicates(['Reviewer', 'Rating', 'Title', 'Body'])

In [ ]:
df.shape

(496, 5)

In [ ]:
def low(text):
  return text.lower()

In [ ]:
from string import punctuation
df['Body'] = df['Body'].apply(lambda x: ''.join(char for char in x if char not in punctuation))
df['Body'] = df['Body'].apply(low)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 496 entries, 0 to 600
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype         
---  ------    --------------  -----         
 0   Reviewer  496 non-null    object        
 1   Rating    496 non-null    float64       
 2   Title     496 non-null    object        
 3   Date      496 non-null    datetime64[ns]
 4   Body      496 non-null    object        
dtypes: datetime64[ns](1), float64(1), object(3)
memory usage: 23.2+ KB


In [ ]:
class nlpreviews:
    def __init__(self):
        import re
        import spacy
        self.spacy = spacy
        self.nlp = spacy.load("en_core_web_sm")

    def aspect_detect(self, text):
        text = text.lower()
        doc = self.nlp(text)
        aspects = []
        for chunk in doc.noun_chunks:
            # Filter out stopwords and lemmatize each token in the chunk
            filtered_tokens = [
                token.lemma_ for token in chunk if not token.is_stop and token.is_alpha
            ]
            if filtered_tokens:  # Add only if there's something left
                aspects.append(' '.join(filtered_tokens))
        return ' '.join(aspects)

In [ ]:
import re

def remove_emojis(text):
    emoji_pattern = re.compile(
        "["
        "\U0001F600-\U0001F64F"  # Emoticons
        "\U0001F300-\U0001F5FF"  # Symbols & pictographs
        "\U0001F680-\U0001F6FF"  # Transport & map symbols
        "\U0001F1E0-\U0001F1FF"  # Flags
        "\U00002500-\U00002BEF"  # Chinese characters
        "\U00002702-\U000027B0"
        "\U00002702-\U000027B0"
        "\U000024C2-\U0001F251"
        "\U0001f926-\U0001f937"
        "\U00010000-\U0010ffff"
        "\u2640-\u2642"
        "\u2600-\u2B55"
        "\u200d"
        "\u23cf"
        "\u23e9"
        "\u231a"
        "\ufe0f"  # Dingbats
        "\u3030"
        "]+",
        flags=re.UNICODE
    )
    return emoji_pattern.sub(r'', text)

In [ ]:
nr = nlpreviews()

In [ ]:
df['Body'] = df['Body'].apply(remove_emojis)

In [ ]:
df['review_key_aspects'] = df['Body'].apply(nr.aspect_detect)

In [ ]:
df.head()

Reviewer  Rating                                              Title  \
0           ROSHAN     5.0                                               Good   
1  P.Kishore Kumar     3.0                                            Nothing   
2           Vikram     1.0                            Do not buy this product   
3           Devika     3.0                                               Good   
4   Mohammad Azhar     1.0  Good night refill stick is broken. Lost of one...   

        Date                                               Body  \
0 2025-06-27                                               good   
1 2025-06-27                                               good   
2 2025-06-21  not happy with product quality very bad smell ...   
3 2025-06-16                            for mosquito repellent…   
4 2025-06-10  product coil broken how to replace in refill c...   

                                  review_key_aspects  
0                                                     
1                                                     
2  product quality good knight year look differen...  
3                                 mosquito repellent  
4  product coil refill customer image customer image

In [ ]:
import nltk
nltk.download('vader_lexicon')  # Download VADER once

[nltk_data] Downloading package vader_lexicon to /root/nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


True

In [ ]:
import spacy
from nltk.sentiment.vader import SentimentIntensityAnalyzer

# Load SpaCy and VADER
nlp = spacy.load("en_core_web_sm")
sid = SentimentIntensityAnalyzer()

def extract_aspect_sentiments(text):
    doc = nlp(text)
    aspect_sentiments = {}

    for sent in doc.sents:
        sentence = sent.text
        parsed_sent = nlp(sentence)
        for token in parsed_sent:
            if token.pos_ == "NOUN":
                # Get modifiers (adjective or adverb)
                modifiers = [child.text for child in token.children if child.dep_ in ("amod", "acomp", "advmod")]
                if modifiers:
                    phrase = f"{' '.join(modifiers)} {token.text}"
                else:
                    phrase = token.text
                sentiment_score = sid.polarity_scores(sentence)["compound"]
                if sentiment_score >= 0.05:
                    sentiment = "positive"
                elif sentiment_score <= -0.05:
                    sentiment = "negative"
                else:
                    sentiment = "neutral"
                aspect_sentiments[token.text.lower()] = sentiment

    return aspect_sentiments

In [ ]:
df["aspect_sentiments"]=df["Body"].apply(extract_aspect_sentiments)

In [ ]:
from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer
from sklearn.decomposition import LatentDirichletAllocation, NMF

In [ ]:
# Count Vectorizer
count_vect = CountVectorizer(stop_words='english')
dtm = count_vect.fit_transform(df['Body'])

# LDA Model
lda = LatentDirichletAllocation(n_components=2, random_state=42)
lda.fit(dtm)

# Topic Display Function
def display_topics(model, feature_names, n_top_words=5):
    for topic_idx, topic in enumerate(model.components_):
        print(f"LDA Topic {topic_idx+1}: " + ", ".join([feature_names[i] for i in topic.argsort()[:-n_top_words - 1:-1]]))

display_topics(lda, count_vect.get_feature_names_out())

LDA Topic 1: good, product, customer, helpful, image
LDA Topic 2: product, price, ok, effective, fake


In [ ]:
import nltk
nltk.download('punkt')
nltk.download('averaged_perceptron_tagger')
nltk.download('stopwords')
nltk.download('omw-1.4')  # Optional: Helps with WordNet if you use lemmatization

[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package averaged_perceptron_tagger to
[nltk_data]     /root/nltk_data...
[nltk_data]   Package averaged_perceptron_tagger is already up-to-
[nltk_data]       date!
[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package omw-1.4 to /root/nltk_data...
[nltk_data]   Package omw-1.4 is already up-to-date!


True

In [ ]:
import os
nltk.data.path.append("/usr/local/nltk_data")

In [ ]:
nltk.download('punkt', download_dir='/usr/local/nltk_data')

[nltk_data] Downloading package punkt to /usr/local/nltk_data...
[nltk_data]   Package punkt is already up-to-date!


True

In [ ]:
import spacy
import nltk
from nltk import pos_tag, word_tokenize
from nltk.corpus import stopwords
nltk.download('punkt')
nltk.download('averaged_perceptron_tagger')
nltk.download('stopwords')

# Load spaCy English model
nlp = spacy.load("en_core_web_sm")

# Example responses
texts = [
    "The product is crunchy and fresh.",
    "I love the eco-friendly and trustworthy packaging.",
    "Feels light and smooth on the skin.",
    "Too expensive for such basic quality.",
    "Incredible taste, but the smell is weird.",
]

[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package averaged_perceptron_tagger to
[nltk_data]     /root/nltk_data...
[nltk_data]   Package averaged_perceptron_tagger is already up-to-
[nltk_data]       date!
[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


In [ ]:
import pandas as pd
import spacy

# Load spaCy model
nlp = spacy.load("en_core_web_sm")

# Sample function
def adjective_detect(text):
    if pd.isna(text):  # handle NaNs
        return ""
    doc = nlp(text)
    adjectives = [token.text for token in doc if token.pos_ == "ADJ"]
    return ' '.join(adjectives)  # Or return adjectives if you want a list

In [ ]:
df["adjective_review"] = df["Body"].apply(adjective_detect)

In [ ]:
import spacy

nlp = spacy.load("en_core_web_sm")

In [ ]:
def ner(text):
  doc = nlp(text)
  for ent in doc.ents:
    return ent.text, ent.label_

In [ ]:
df["ner_review"] = df["Body"].apply(ner)

In [ ]:
df.head()

Reviewer  Rating                                              Title  \
0           ROSHAN     5.0                                               Good   
1  P.Kishore Kumar     3.0                                            Nothing   
2           Vikram     1.0                            Do not buy this product   
3           Devika     3.0                                               Good   
4   Mohammad Azhar     1.0  Good night refill stick is broken. Lost of one...   

        Date                                               Body  \
0 2025-06-27                                               good   
1 2025-06-27                                               good   
2 2025-06-21  not happy with product quality very bad smell ...   
3 2025-06-16                            for mosquito repellent…   
4 2025-06-10  product coil broken how to replace in refill c...   

                                  review_key_aspects  \
0                                                      
1                                                      
2  product quality good knight year look differen...   
3                                 mosquito repellent   
4  product coil refill customer image customer image   

                                   aspect_sentiments  \
0                                                 {}   
1                                                 {}   
2  {'product': 'positive', 'quality': 'negative',...   
3                           {'repellent': 'neutral'}   
4  {'product': 'negative', 'coil': 'negative', 'r...   

                adjective_review             ner_review  
0                           good                   None  
1                           good                   None  
2  happy bad good last different  (last 10 years, DATE)  
3                                                  None  
4                                                  None

In [ ]:
from transformers import AutoTokenizer, AutoModelForSequenceClassification
import torch
import torch.nn.functional as F

model_name = "bhadresh-savani/bert-base-go-emotion"
tokenizer = AutoTokenizer.from_pretrained(model_name)
model = AutoModelForSequenceClassification.from_pretrained(model_name)

In [ ]:
def predict_emotions(text, top_k=3):
    # Tokenize the input text
    inputs = tokenizer(text, return_tensors="pt", truncation=True)

    # Get model predictions
    with torch.no_grad():
        logits = model(**inputs).logits
        probs = torch.sigmoid(logits)[0]  # Apply sigmoid for multi-label classification

    # Get top k emotions
    top_probs, top_indices = torch.topk(probs, k=top_k)
    labels = [model.config.id2label[idx.item()] for idx in top_indices]
    scores = [prob.item() for prob in top_probs]

    # Return results as a list of tuples (label, score)
    return list(zip(labels, scores))

In [ ]:
def emotion(text):
  abc = predict_emotions(text)  # Use the function parameter 'text'
  return abc

df["emotions_review"] = df["Body"].apply(emotion)

In [ ]:
from transformers import pipeline

# Load a general emotion classifier that includes empathy-related labels
classifier = pipeline("text-classification", model="j-hartmann/emotion-english-distilroberta-base", top_k=None)

Device set to use cpu


In [ ]:
def empathy(text):
  results = classifier(text)
  return results
df["empathy_review"] = df["Body"].apply(empathy)

In [ ]:
from nltk.sentiment.vader import SentimentIntensityAnalyzer
import nltk

# Download VADER lexicon (only once)
nltk.download('vader_lexicon')

# Initialize the VADER sentiment analyzer
sid = SentimentIntensityAnalyzer()

# Define function to get sentiment label
def get_sentiment(text):
    if not isinstance(text, str) or text.strip() == "":
        return "neutral"  # default for empty or invalid entries
    score = sid.polarity_scores(text)["compound"]
    if score >= 0.05:
        return "positive"
    elif score <= -0.05:
        return "negative"
    else:
        return "neutral"

# Apply to 'review' column
df['predicted_sentiment'] = df['Body'].apply(get_sentiment)


[nltk_data] Downloading package vader_lexicon to /root/nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


In [ ]:
df.head()

Reviewer  Rating                                              Title  \
0           ROSHAN     5.0                                               Good   
1  P.Kishore Kumar     3.0                                            Nothing   
2           Vikram     1.0                            Do not buy this product   
3           Devika     3.0                                               Good   
4   Mohammad Azhar     1.0  Good night refill stick is broken. Lost of one...   

        Date                                               Body  \
0 2025-06-27                                               good   
1 2025-06-27                                               good   
2 2025-06-21  not happy with product quality very bad smell ...   
3 2025-06-16                            for mosquito repellent…   
4 2025-06-10  product coil broken how to replace in refill c...   

                                  review_key_aspects  \
0                                                      
1                                                      
2  product quality good knight year look differen...   
3                                 mosquito repellent   
4  product coil refill customer image customer image   

                                   aspect_sentiments  \
0                                                 {}   
1                                                 {}   
2  {'product': 'positive', 'quality': 'negative',...   
3                           {'repellent': 'neutral'}   
4  {'product': 'negative', 'coil': 'negative', 'r...   

                adjective_review             ner_review  \
0                           good                   None   
1                           good                   None   
2  happy bad good last different  (last 10 years, DATE)   
3                                                  None   
4                                                  None   

                                     emotions_review  \
0  [(admiration, 0.5878942608833313), (neutral, 0...   
1  [(admiration, 0.5878942608833313), (neutral, 0...   
2  [(neutral, 0.270477294921875), (disapproval, 0...   
3  [(neutral, 0.5902127623558044), (annoyance, 0....   
4  [(neutral, 0.2834271192550659), (disappointmen...   

                                      empathy_review predicted_sentiment  
0  [[{'label': 'neutral', 'score': 0.632514417171...            positive  
1  [[{'label': 'neutral', 'score': 0.632514417171...            positive  
2  [[{'label': 'sadness', 'score': 0.895039975643...            positive  
3  [[{'label': 'neutral', 'score': 0.793119966983...             neutral  
4  [[{'label': 'neutral', 'score': 0.530444085597...            negative

In [ ]:
def rating_sentiment(rating):
    if rating >= 4.0:
        return "positive"
    elif rating <= 2.0:
        return "negative"
    else:
        return "neutral"

df['actual_sentiment'] = df['Rating'].apply(rating_sentiment)

In [ ]:
df[['actual_sentiment','Rating']].sample(10)

actual_sentiment  Rating
144         negative     1.0
540         positive     5.0
59          positive     5.0
50          positive     5.0
264         negative     2.0
186         negative     1.0
252         negative     2.0
153         negative     1.0
493         positive     4.0
202         negative     2.0

In [ ]:
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

In [ ]:
print(accuracy_score(df['actual_sentiment'], df['predicted_sentiment']))
print(confusion_matrix(df['actual_sentiment'], df['predicted_sentiment']))
print(classification_report(df['actual_sentiment'], df['predicted_sentiment']))

0.6108870967741935
[[ 96  44  59]
 [ 18  31  49]
 [  5  18 176]]
              precision    recall  f1-score   support

    negative       0.81      0.48      0.60       199
     neutral       0.33      0.32      0.32        98
    positive       0.62      0.88      0.73       199

    accuracy                           0.61       496
   macro avg       0.59      0.56      0.55       496
weighted avg       0.64      0.61      0.60       496



In [ ]:
df.shape

(496, 13)

In [ ]:
import pandas as pd
from nltk.sentiment.vader import SentimentIntensityAnalyzer
import nltk

# Download VADER lexicon if not already downloaded
nltk.download('vader_lexicon')

# Initialize the sentiment analyzer
sid = SentimentIntensityAnalyzer()

# Compute compound sentiment score (-1 to 1)
df['computed_sentiment_score'] = df['Body'].apply(lambda x: sid.polarity_scores(x)['compound'])

[nltk_data] Downloading package vader_lexicon to /root/nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


In [ ]:
df[['actual_sentiment','computed_sentiment_score']].sample(10)

actual_sentiment  computed_sentiment_score
547         positive                    0.9800
315          neutral                   -0.8367
380          neutral                    0.4404
488         positive                    0.4404
600         positive                    0.0000
595         positive                    0.3612
484         positive                    0.9306
375          neutral                    0.0000
444         positive                    0.0000
318          neutral                    0.0000

In [ ]:
for i in range(1,31):
  def extrcat_sentiment(score):
    if score >= 0.01*i:
        return "positive"
    elif score <= -0.01*i:
        return "negative"
    else:
        return "neutral"

  df[f'computed_sentiment_{i}'] = df['computed_sentiment_score'].apply(extrcat_sentiment)
  print(f"\nReports when threshold = {i*0.01}")
  print(accuracy_score(df['actual_sentiment'], df[f'computed_sentiment_{i}']))
  print(confusion_matrix(df['actual_sentiment'], df[f'computed_sentiment_{i}']))
  print(classification_report(df['actual_sentiment'], df[f'computed_sentiment_{i}']))


Reports when threshold = 0.01
0.6129032258064516
[[ 97  40  62]
 [ 18  31  49]
 [  5  18 176]]
              precision    recall  f1-score   support

    negative       0.81      0.49      0.61       199
     neutral       0.35      0.32      0.33        98
    positive       0.61      0.88      0.72       199

    accuracy                           0.61       496
   macro avg       0.59      0.56      0.55       496
weighted avg       0.64      0.61      0.60       496


Reports when threshold = 0.02
0.6129032258064516
[[ 97  40  62]
 [ 18  31  49]
 [  5  18 176]]
              precision    recall  f1-score   support

    negative       0.81      0.49      0.61       199
     neutral       0.35      0.32      0.33        98
    positive       0.61      0.88      0.72       199

    accuracy                           0.61       496
   macro avg       0.59      0.56      0.55       496
weighted avg       0.64      0.61      0.60       496


Reports when threshold = 0.03
0.6129032258064



---


Reports when threshold = 0.03

0.6129032258064516

[[ 97  40  62]

 [ 18  31  49]

 [  5  18 176]]

              precision    recall  f1-score   support

    negative       0.81      0.49      0.61       199
     neutral       0.35      0.32      0.33        98
    positive       0.61      0.88      0.72       199

    accuracy                           0.61       496
   macro avg       0.59      0.56      0.55       496
weighted avg       0.64      0.61      0.60       496

In [ ]:
for i in range(1,31):
  df.drop(f'computed_sentiment_{i}', axis=1, inplace=True)

In [ ]:
def extrcat_sentiment(score):
    if score >= 0.03:
        return "positive"
    elif score <= -0.03:
        return "negative"
    else:
        return "neutral"

In [ ]:
df['computed_sentiment'] = df['computed_sentiment_score'].apply(extrcat_sentiment)

In [ ]:
df.head()

Reviewer  Rating                                              Title  \
0           ROSHAN     5.0                                               Good   
1  P.Kishore Kumar     3.0                                            Nothing   
2           Vikram     1.0                            Do not buy this product   
3           Devika     3.0                                               Good   
4   Mohammad Azhar     1.0  Good night refill stick is broken. Lost of one...   

        Date                                               Body  \
0 2025-06-27                                               good   
1 2025-06-27                                               good   
2 2025-06-21  not happy with product quality very bad smell ...   
3 2025-06-16                            for mosquito repellent…   
4 2025-06-10  product coil broken how to replace in refill c...   

                                  review_key_aspects  \
0                                                      
1                                                      
2  product quality good knight year look differen...   
3                                 mosquito repellent   
4  product coil refill customer image customer image   

                                   aspect_sentiments  \
0                                                 {}   
1                                                 {}   
2  {'product': 'positive', 'quality': 'negative',...   
3                           {'repellent': 'neutral'}   
4  {'product': 'negative', 'coil': 'negative', 'r...   

                adjective_review             ner_review  \
0                           good                   None   
1                           good                   None   
2  happy bad good last different  (last 10 years, DATE)   
3                                                  None   
4                                                  None   

                                     emotions_review  \
0  [(admiration, 0.5878942608833313), (neutral, 0...   
1  [(admiration, 0.5878942608833313), (neutral, 0...   
2  [(neutral, 0.270477294921875), (disapproval, 0...   
3  [(neutral, 0.5902127623558044), (annoyance, 0....   
4  [(neutral, 0.2834271192550659), (disappointmen...   

                                      empathy_review predicted_sentiment  \
0  [[{'label': 'neutral', 'score': 0.632514417171...            positive   
1  [[{'label': 'neutral', 'score': 0.632514417171...            positive   
2  [[{'label': 'sadness', 'score': 0.895039975643...            positive   
3  [[{'label': 'neutral', 'score': 0.793119966983...             neutral   
4  [[{'label': 'neutral', 'score': 0.530444085597...            negative   

  actual_sentiment  computed_sentiment_score computed_sentiment  
0         positive                    0.4404           positive  
1          neutral                    0.4404           positive  
2         negative                    0.2034           positive  
3          neutral                    0.0000            neutral  
4         negative                   -0.4767           negative

In [ ]:
df.to_csv('good_night_analysed_reviews.csv',index=False)